### PR TITLE
STD-7 | Domain models (code)

### PR DESCRIPTION
**Pull request by zefirlover**

### CODE REVIEW

File: StudySharp.Domain/Models/Learning/Content.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public class Content
+    {
+        public int Id { get; set; }
+        public Course Course { get; set; }
+        
+        public List<TheoryBlock> TheoryBlocks { get; set; }
+        public TheoryBlock TheoryBlock { get; set; }
+        
+        public List<PracticalBlock> PracticalBlocks { get; set; }
+        public PracticalBlock PracticalBlock { get; set; }
```
Comment (SleepyF0X) [inline StudySharp.Domain/Models/Learning/Content.cs]: For what do you use a Collection of models and a single model?

File: StudySharp.Domain/Models/Learning/Course.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace StudySharp.Domain.Models.Learning
+{
+    public class Course
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public List<Tag> Tags { get; set; }
+        public DateTimeOffset CreatedAt { get; set; }
+        public DateTimeOffset RedactedAt { get; set; }
```
Comment (SleepyF0X) [inline StudySharp.Domain/Models/Learning/Course.cs]: Its KolaCation idea?

File: StudySharp.Domain/Models/Learning/Quiz.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace StudySharp.Domain.Models.Learning
+{
+    public class Quiz
+    {
+        public int Id { get; set; }
+        public PracticalBlock PracticalBlock { get; set; }
```
Comment (SleepyF0X) [inline StudySharp.Domain/Models/Learning/Quiz.cs:9]: How do you get a PracticalBlock model without PracticalBlockId?

File: StudySharp.DomainServices/Configurations/TeacherConfiguration.cs [hunk 4]
```diff
@@ -12,7 +12,11 @@ public void Configure(EntityTypeBuilder<Teacher> builder)
...
             builder
-                .Property(_ => _.Name)
+                .Property(_ => _.FirstName)
+                .IsRequired();
+
+            builder
+                .Property(_ => _.LastName)
                 .IsRequired();
         }
     }
```
Comment (SleepyF0X) [inline StudySharp.DomainServices/Configurations/TeacherConfiguration.cs:22]: Please, add similar configurations for another models

File: StudySharp.Domain/Models/Learning/Quiz.cs (hunk 3 above)
Comment (zefirlover) [reply StudySharp.Domain/Models/Learning/Quiz.cs:9]: Fixed

File: StudySharp.Domain/Models/Learning/Course.cs (hunk 2 above)
Comment (zefirlover) [reply StudySharp.Domain/Models/Learning/Course.cs]: It WAS
Yes, it was

File: StudySharp.Domain/Models/Learning/Content.cs (hunk 1 above)
Comment (zefirlover) [reply StudySharp.Domain/Models/Learning/Content.cs]: Kola said that we need them both
I doubted, but add it like he said

Comment (zefirlover) [reply StudySharp.Domain/Models/Learning/Content.cs]: Models was remowed

File: StudySharp.Domain/Models/Learning/Content.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+
+namespace StudySharp.Domain.Models.Learning
+{
+    public class Content
+    {
+        public int Id { get; set; }
+        public Course Course { get; set; }
```
Comment (SleepyF0X) [inline StudySharp.Domain/Models/Learning/Content.cs:10]: How do you get a Course model without CourseId?

@KolaCation we really need Id here?

File: StudySharp.DomainServices/Configurations/CourseConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                .IsRequired();
+
+            builder
+                .Property(_ => _.CreatedAt)
+                .IsRequired();
+
+            builder
+                .Property(_ => _.RedactedAt)
+                .IsRequired();
+        }
```
Comment (SleepyF0X) [inline StudySharp.DomainServices/Configurations/CourseConfiguration.cs]: Not required field

File: StudySharp.Domain/Models/Learning/Course.cs (hunk 2 above)
Comment (SleepyF0X) [reply StudySharp.Domain/Models/Learning/Course.cs]: Ok, set RedactedAt nullable

File: StudySharp.Domain/Models/Learning/Content.cs (hunk 5 above)
Comment (zefirlover) [reply StudySharp.Domain/Models/Learning/Content.cs:10]: For the 1 to 1 connection we need it

**Review by SleepyF0X — APPROVED** [review]

**Pull request merged**